crates/ra_ide/src/inlay_hints.rs

```diff
@@ -245,14 +244,32 @@ fn should_show_param_hint(
     } else {
         fn_signature.parameters.len()
     };
+
     // avoid displaying hints for common functions like map, filter, etc.
-    if parameters_len == 1 && (param_name.len() == 1 || param_name == "predicate") {
+    // or other obvious words used in std
+    if parameters_len == 1 && is_obvious_param(param_name) {
         return false;
     }
-
     true
 }
 
+fn is_argument_similar_to_param(argument: &ast::Expr, param_name: &str) -> bool {
+    let argument_string = if let ast::Expr::RefExpr(ref_expr) = argument {
+        ref_expr.syntax().last_token().expect("RefExpr should have a last_token").to_string()
```

Comment:
This expect can in theory fire. I suggest doing this:

```
fn remove_ref(expr: ast::Expr) -> ast::Expr {
    if let ast::Expr::RefExpr(ref_expr) = &expr {
        if let Some(inner) = ref_expr.expr() {
            return expr;
        }
    }
    expr
}
```

Reply:
I didn't realize there was a `.expr()` on the RefExpr. I'll fix that!